article-analyzer/src/main/java/co/inventorsoft/academy/spring/repositories/ArticleRepository.java

```diff
@@ -0,0 +1,51 @@
+package co.inventorsoft.academy.spring.repositories;
+
+import co.inventorsoft.academy.spring.models.Article;
+import co.inventorsoft.academy.spring.utils.UUIDDeserializer;
+import co.inventorsoft.academy.spring.utils.UUIDSerializer;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Repository;
+
+
+/**
+ * Data access layer for Article entity.
+ */
+@Repository
+//public interface ArticleRepository extends JpaRepository<Article, UUID> {
+public class ArticleRepository {
+    @Value("${ARTICLES_FILE}")
+    String articlesFile;
+    private final Gson gson;
+
+    @Autowired
+    public ArticleRepository() {
+        this.gson = new GsonBuilder()
+            .registerTypeAdapter(UUID.class, new UUIDDeserializer())
+            .registerTypeAdapter(UUID.class, new UUIDSerializer())
+            .create();
```

Comment:
why these serializer/deserializer is needed?

Reply:
I used UUID type and gson didnt work well with it so i used custom serializer and deserializer, but it wasn't problem with them and when i solved it completely forgot about this custom stuff, guess i should remove them